### ISSUE TITLE
Add support for Google Authenticator / TOTP

### ISSUE THREAD

**Issue by sffc** [octave-online/octave-online-server#5]
This should be added as another login strategy for easy use with smartphones.

Example: https://github.com/jaredhanson/passport-totp

**Comment by Abhirup-99** [comment]
@sffc is this issue still up? I would like to work on this.

**Comment by sffc** [comment]
Yep, this issue is still open.  The problem is that today, accounts are identified by their email.  If we add TOTP as an alternative to email-based login (including Google OAuth), we need to figure out how to identify those accounts.

**Comment by Abhirup-99** [comment]
In TOTP, after the authentication, we may ask the user for one-time email verification the first time, then the email is authenticated and the user can continue with TOTP for any future session.

**Comment by Abhirup-99** [comment]
On a side note: I am using octave online for my classes, would be glad to push code over here. This is an awesome platform, thanks for this.
